apis/projectcontour/v1/httpproxy.go

```diff
@@ -320,6 +323,48 @@ type Route struct {
 	// Rewriting the 'Host' header is not supported.
 	// +optional
 	ResponseHeadersPolicy *HeadersPolicy `json:"responseHeadersPolicy,omitempty"`
+	// The policy for rate limiting on the route.
+	// +optional
+	RateLimitPolicy *RateLimitPolicy `json:"rateLimitPolicy,omitempty"`
+}
+
+// RateLimitPolicy defines rate limiting parameters.
+type RateLimitPolicy struct {
+	// Local defines local rate limiting parameters, i.e. parameters
+	// for rate limiting that occurs within each Envoy pod as requests
+	// are handled.
+	Local *LocalRateLimitPolicy `json:"local,omitempty"`
+}
+
+// LocalRateLimitPolicy defines local rate limiting parameters.
+type LocalRateLimitPolicy struct {
+	// Requests defines how many requests per unit of time should
+	// be allowed before rate limiting occurs.
+	// +required
+	// +kubebuilder:validation:Minimum=1
+	Requests uint32 `json:"requests"`
```

Comment:
Would it make sense to combine `Requests` & `Unit` into a duration field so you could specify it once like: `1m`?

Reply:
It could work, but it'd be a little weird because the relationship is "n <requests> / 1 <unit>", so if you merged it into a duration string, you'd write e.g. `30s` to mean "30 requests per second". Personally I'd rather keep it as-is. We could rename `Requests` to `RequestsPerUnit` or `Rate` or something else.

Reply:
nah your design is correct, I wasn't thinking about it correctly.

Reply:
At one point we talked about (or maybe it was just in my head), of coming up with all the error logic to figure out what's wrong with stuff, then can return there before you get too deep into the logic, but yeah it gets messy quickly.